internal/clients/http/sources/sources_client.go

```diff
@@ -98,8 +101,11 @@ func (c *sourcesClient) ListProvisioningSourcesByProvider(ctx context.Context, p
 		return nil, fmt.Errorf("failed to get provider name according to sources service: %w", err)
 	}
 
-	resp, err := c.client.ListApplicationTypeSourcesWithResponse(ctx, appTypeId, &ListApplicationTypeSourcesParams{}, headers.AddSourcesIdentityHeader,
-		headers.AddEdgeRequestIdHeader, BuildQuery("filter[source_type][name]", sourcesProviderName))
+	offset := strconv.Itoa(pagination.GetOffset(ctx))
+	limit := strconv.Itoa(pagination.GetLimit(ctx))
```

Comment:
Might be pretty useful to create `pagination.OffsetString(ctx)` doing conversion for you. Also pointer (below). Just an idea.